assets/tests/components/nav.test.tsx

```diff
@@ -0,0 +1,52 @@
+import React from "react"
+import { BrowserRouter } from "react-router-dom"
+import renderer from "react-test-renderer"
+import appData from "../../src/appData"
+import { Nav } from "../../src/components/nav"
+import featureIsEnabled from "../../src/laboratoryFeatures"
+import { OpenView } from "../../src/state"
+
+jest.mock("../../src/laboratoryFeatures", () => ({
+  __esModule: true,
+  default: jest.fn(() => false),
+}))
+
+jest.mock("../../src/appData", () => ({
+  __esModule: true,
+  default: jest.fn(() => false),
+}))
+
+describe("Nav", () => {
+  test("renders children with TabBar", () => {
+    const tree = renderer
+      .create(
+        <BrowserRouter>
+          <Nav pickerContainerIsVisible={true} openView={OpenView.None}>
+            Hello, world!
+          </Nav>
+        </BrowserRouter>
+      )
+      .toJSON()
+    expect(tree).toMatchSnapshot()
```

Comment:
This is a very big snapshot. It's too big to read by hand and spot changes or errors in, and it'll be a chore to keep updated. Instead, you could make some assertions about what you expect to be in there. e.g. "expect the tree to have the content "Hello, world".", or "expect this to contain an element with the class m-tab-bar".

I know a big testing rewrite to address all of these big snapshots in the codebase is probably too big to be worth it, but it'd be nice to use this approach in new code.

I'm less concerned about getting the testing framework right or using best practices as I am about writing down what's significant that this test is supposed to be checking for.

(Same for the test below. That should specifically assert that the late view icon exists.)

Reply:
Okay, so I did some cleanup here to use React Testing Library and assert the important things rather than using snapshots. Note that RTL is rather opinionated about how to select things in order to encourage you to write good HTML, but the tab bar is old and doesn't really adhere to best practices, so the best way to select the late view icon is to use the `data-testid` escape hatch. This means in turn that it changes some _other_ snapshots, but I think on net this is the least bad thing. Also the existing `TabBar` component is going away once we release the new navigation so we'll be able to get rid of those old snapshots anyway.